src/AnalysisDD4hep.cxx

```diff
@@ -0,0 +1,735 @@
+	  clus->x = HcalBarrelClusters_x[icl]; // use edep?
+	  clus->y = HcalBarrelClusters_y[icl]; // use edep?
+	  clus->z = HcalBarrelClusters_z[icl]; // use edep?
+	  clus->theta = HcalBarrelClusters_theta[icl]; // use edep?
+	  clus->phi = HcalBarrelClusters_phi[icl]; // use edep?
+	  v_hcal_clusters.push_back(clus);
+	}
+      // HcalEndcapP
+      for(int icl=0; icl<HcalEndcapPClusters_x.GetSize(); icl++)
+	{
+	  Clusters* clus = new Clusters();
+	  clus->E = HcalEndcapPClusters_energy[icl]; // use edep?
+	  clus->x = HcalEndcapPClusters_x[icl]; // use edep?
+	  clus->y = HcalEndcapPClusters_y[icl]; // use edep?
+	  clus->z = HcalEndcapPClusters_z[icl]; // use edep?
+	  clus->theta = HcalEndcapPClusters_theta[icl]; // use edep?
+	  clus->phi = HcalEndcapPClusters_phi[icl]; // use edep?
+	  v_hcal_clusters.push_back(clus);
+	}
+
+      // HcalEndcapN
+      for(int icl=0; icl<HcalEndcapNClusters_x.GetSize(); icl++)
+	{
+	  Clusters* clus = new Clusters();
+	  clus->E = HcalEndcapNClusters_energy[icl]; // use edep?
+	  clus->x = HcalEndcapNClusters_x[icl]; // use edep?
+	  clus->y = HcalEndcapNClusters_y[icl]; // use edep?
+	  clus->z = HcalEndcapNClusters_z[icl]; // use edep?
+	  clus->theta = HcalEndcapNClusters_theta[icl]; // use edep?
+	  clus->phi = HcalEndcapNClusters_phi[icl]; // use edep?
+	  v_hcal_clusters.push_back(clus);
+	}
+
+      // Barrel
+      for(int icl=0; icl<EcalBarrelClusters_x.GetSize(); icl++)
+	{
+	  Clusters* clus = new Clusters();
+	  clus->E = EcalBarrelClusters_energy[icl]; // use edep?
+	  clus->x = EcalBarrelClusters_x[icl]; // use edep?
+	  clus->y = EcalBarrelClusters_y[icl]; // use edep?
+	  clus->z = EcalBarrelClusters_z[icl]; // use edep?
+	  clus->theta = EcalBarrelClusters_theta[icl]; // use edep?
+	  clus->phi = EcalBarrelClusters_phi[icl]; // use edep?
+	  v_ecal_clusters.push_back(clus);
+	}
+
+      // EndcapP
+      for(int icl=0; icl<EcalEndcapPClusters_x.GetSize(); icl++)
+	{
+	  Clusters* clus = new Clusters();
+	  clus->E = EcalEndcapPClusters_energy[icl]; // use edep?
+	  clus->x = EcalEndcapPClusters_x[icl]; // use edep?
+	  clus->y = EcalEndcapPClusters_y[icl]; // use edep?
+	  clus->z = EcalEndcapPClusters_z[icl]; // use edep?
+	  clus->theta = EcalEndcapPClusters_theta[icl]; // use edep?
+	  clus->phi = EcalEndcapPClusters_phi[icl]; // use edep?
+	  v_ecal_clusters.push_back(clus);
+	}
+
+      // EndcapN
+      for(int icl=0; icl<EcalEndcapNClusters_x.GetSize(); icl++)
+	{
+	  Clusters* clus = new Clusters();
+	  clus->E = EcalEndcapNClusters_energy[icl]; // use edep?
+	  clus->x = EcalEndcapNClusters_x[icl]; // use edep?
+	  clus->y = EcalEndcapNClusters_y[icl]; // use edep?
+	  clus->z = EcalEndcapNClusters_z[icl]; // use edep?
+	  clus->theta = EcalEndcapNClusters_theta[icl]; // use edep?
+	  clus->phi = EcalEndcapNClusters_phi[icl]; // use edep?
+	  v_ecal_clusters.push_back(clus);
+	}
+
+      // calculate true DIS kinematics
+      kinTrue->CalculateDISbyElectron(); // generated (truth)
```

Comment:
should this be moved to be between the `mcparticles` loop and the calorimeters loops (L372)?

Reply:
fixed by 23d4481